redhat-developer/web-terminal-tooling
# Generate versions at build

### Description
To support installing CLI tooling via packages instead of by direct downloads, generate the "installed tooling" help doc at build time rather than hard-coding a set of environment variables. This ensures the correct, installed version for all tooling is printed when 'help' is typed in the terminal.

For Kustomize, the downstream image no longer has a separate binary for kustomize installed. In this case, the version from `kubectl` is printed, with a note.

In addition, clean up some leftovers from earlier versions:
* Move setting PS1 into initial .bashrc instead of entrypoint.sh (holdover from before tooling image had built-in .bashrc)
* Generate odo completions at build time (odo now has a `odo completions bash` command)

### How to test
```
./build.sh -i <IMAGE+TAG> -u
podman run -it --rm <IMAGE+TAG> /bin/bash
```
Verify that 
a) Typing `help` lists installed versions for installed CLIs
b) Each installed CLI's actual version matches output of `help`
